Unable to build BLAS - complex.h missing

Hello, 

I am currently trying to build BLAS as it is needed for another code I will be using - NEML (https://github.com/Argonne-National-Laboratory/neml). My current setup is Win 10, Visual Studio 2012 with installed Intel Parallel 2015. 

I have followed the steps for installing OpenBlas from https://github.com/xianyi/OpenBLAS/wiki/How-to-use-OpenBLAS-in-Microsoft-Visual-Studio, but the Conda Prompt exited with an error during the build - `ninja: build stopped: subcommand failed.` ; see attached output from the shell. It appears that build encountered an error in not finding complex.h. 
Would it be possible to get some feedback on what the issue might be and how it could be resolved? I can provide any info needed to help resolve this. Sadly I am restricted to using Windows because of other softwares which will later use the above NEML model which requires BLAS. 
Many thanks all
[Anaconda Prmt - BLAS.txt](https://github.com/xianyi/OpenBLAS/files/6474256/Anaconda.Prmt.-.BLAS.txt)

Ok, I installed chocolatey and then installed make. After that, I opened cmd, `cd` to \OpenBLAS and `make` . The result does not seem correct.. : 
```
D:\GITlib\OpenBLAS>make
process_begin: CreateProcess(NULL, uname -m, ...) failed.
Makefile.system:14: pipe: No error
The system cannot find the path specified.
process_begin: CreateProcess(NULL, uname -s, ...) failed.
Makefile.system:56: pipe: Bad file descriptor
process_begin: CreateProcess(NULL, gcc -DGEMM_MULTITHREAD_THRESHOLD=4 -DNO_PARALLEL_MAKE=0 -o getarch getarch.c cpuid.S, ...) failed.
make (e=2): The system cannot find the file specified.
make: *** [Makefile.prebuild:70: getarch] Error 2
process_begin: CreateProcess(NULL, uname -m, ...) failed.
```
I guess I am missing something?

After removing `<complex.h>` from cdot.c, the build continued up to [12718/17241] but failed again due with the following:
```
[12718/17241] Building C object utest\CMakeFiles\openblas_utest.dir\test_kernel_regress.c.obj
FAILED: utest/CMakeFiles/openblas_utest.dir/test_kernel_regress.c.obj
C:\Users\Mark\miniconda3\Library\bin\clang-cl.exe  /nologo  -I..\lapack-netlib\LAPACKE\include -I..\ -I. /DWIN32 /D_WINDOWS /W3  -DHAVE_C11 -DMS_ABI -Wall -m64 -DF_INTERFACE_FLANG -DDYNAMIC_ARCH -DSMP_SERVER -DMAX_CPU_NUMBER=8 -DMAX_PARALLEL_NUMBER=1 -DNO_AFFINITY -DVERSION="\"0.3.15.dev\"" -DBUILD_SINGLE -DBUILD_DOUBLE -DBUILD_COMPLEX -DBUILD_COMPLEX16 /MD /O2 /Ob2 /DNDEBUG /Zi /showIncludes /Foutest\CMakeFiles\openblas_utest.dir\test_kernel_regress.c.obj /Fdutest\CMakeFiles\openblas_utest.dir\ -c -- ..\utest\test_kernel_regress.c
In file included from ..\utest\test_kernel_regress.c:1:
..\utest/openblas_utest.h(35,9) : warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _OPENBLAS_UTEST_H_
        ^
In file included from ..\utest\test_kernel_regress.c:1:
In file included from ..\utest/openblas_utest.h:38:
In file included from ..\utest/ctest.h:31:
C:\Users\Mark\miniconda3\Library\lib\clang\11.0.1\include\inttypes.h(18,2) : error: MSVC does not have inttypes.h prior to Visual Studio 2013
#error MSVC does not have inttypes.h prior to Visual Studio 2013
 ^
C:\Users\Mark\miniconda3\Library\lib\clang\11.0.1\include\inttypes.h(21,15) : fatal error: 'inttypes.h' file not found
#include_next <inttypes.h>
              ^~~~~~~~~~~~
1 warning and 2 errors generated.
```
...
```
[12727/17241] Generating Fortran dyndep file CMakeFiles\openblas.dir\Fortran.dd
ninja: build stopped: subcommand failed.
```
It appears that it has an issue with the version of VS - do you think there is way to bypass this as for `complex.h`.  Alternatively, I could try using the `make` approach. Would this require adding an env variable CMAKE_C_COMPILER to Intel C compiler?
To answer your question Martin, I am needing NEML (and respectively BLAS) for the structural solver Ansys. Ansys versions only accept specific versions of VS and Intel Parallel, and as I have an old version of Ansys currently I have to stick to VS2012. Thanks!

Update: It did build and install with your suggestion Martin; thank you! I can see however that when trying to compile the NEML software, it does not find the BLAS library. Is there a need to define any env var to the OpenBLAS folder so that `cmake` recognises it? Cmake documentation tells me it should recognise OpenBLAS without any additional work. I will probably raise a question to NEML folks on their git for this and close this issue as solved here. 

```
CMake Error at D:/GITlib/cmake/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find BLAS (missing: BLAS_LIBRARIES)
```

--- a/kernel/x86_64/cdot.c
+++ b/kernel/x86_64/cdot.c
@@ -27,7 +27,6 @@ USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
 #include "common.h"
-#include <complex.h>
 
 
 #if defined(BULLDOZER)

--- a/utest/ctest.h
+++ b/utest/ctest.h
@@ -28,7 +28,10 @@
 #define WEAK
 #endif
 
+#ifndef __MSC_VER
 #include <inttypes.h> /* intmax_t, uintmax_t, PRI* */
+#endif
+
 #include <stddef.h> /* size_t */
 
 typedef void (*SetupFunc)(void*);
@@ -71,6 +74,13 @@ struct ctest {
 
 #define __CTEST_NO_TIME
 #define CTEST_NO_COLORS
+
+#if __MSC_VER >= 1500
+#include <inttypes.h>
+#else
+#include <stdint.h>
+#define CTEST_NO_INTTYPES
+#endif
 
 #ifndef CTEST_ADD_TESTS_MANUALLY
 #pragma section(".ctest$a")
@@ -480,11 +490,19 @@ void assert_data(const unsigned char* exp, size_t expsize,
                  const char* caller, int line) {
     size_t i;
     if (expsize != realsize) {
+#ifndef CTEST_NO_INTTYPES
         CTEST_ERR("%s:%d  expected %" PRIuMAX " bytes, got %" PRIuMAX, caller, line, (uintmax_t) expsize, (uintmax_t) realsize);
+#else
+        CTEST_ERR("%s:%d  expected %u bytes, got %u", caller, line, (uintmax_t) expsize, (uintmax_t) realsize);
+#endif
     }
     for (i=0; i<expsize; i++) {
         if (exp[i] != real[i]) {
+#ifndef CTEST_NO_INTTYPES
             CTEST_ERR("%s:%d expected 0x%02x at offset %" PRIuMAX " got 0x%02x",
+#else
+            CTEST_ERR("%s:%d expected 0x%02x at offset %u got 0x%02x",
+#endif
                 caller, line, exp[i], (uintmax_t) i, real[i]);
         }
     }
@@ -492,31 +510,51 @@ void assert_data(const unsigned char* exp, size_t expsize,
 
 void assert_equal(intmax_t exp, intmax_t real, const char* caller, int line) {
     if (exp != real) {
+#ifndef CTEST_NO_INTTYPES
         CTEST_ERR("%s:%d  expected %" PRIdMAX ", got %" PRIdMAX, caller, line, exp, real);
+#else
+        CTEST_ERR("%s:%d  expected %d, got %d", caller, line, exp, real);
+#endif
     }
 }
 
 void assert_equal_u(uintmax_t exp, uintmax_t real, const char* caller, int line) {
     if (exp != real) {
+#ifndef CTEST_NO_INTTYPES
         CTEST_ERR("%s:%d  expected %" PRIuMAX ", got %" PRIuMAX, caller, line, exp, real);
+#else
+        CTEST_ERR("%s:%d  expected %u, got %u", caller, line, exp, real);
+#endif
     }
 }
 
 void assert_not_equal(intmax_t exp, intmax_t real, const char* caller, int line) {
     if ((exp) == (real)) {
+#ifndef CTEST_NO_INTTYPES
         CTEST_ERR("%s:%d  should not be %" PRIdMAX, caller, line, real);
+#else
+        CTEST_ERR("%s:%d  should not be %d", caller, line, real);
+#endif
     }
 }
 
 void assert_not_equal_u(uintmax_t exp, uintmax_t real, const char* caller, int line) {
     if ((exp) == (real)) {
+#ifndef CTEST_NO_INTTYPES
         CTEST_ERR("%s:%d  should not be %" PRIuMAX, caller, line, real);
+#else
+        CTEST_ERR("%s:%d  should not be %u", caller, line, real);
+#endif
     }
 }
 
 void assert_interval(intmax_t exp1, intmax_t exp2, intmax_t real, const char* caller, int line) {
     if (real < exp1 || real > exp2) {
+#ifndef CTEST_NO_INTTYPES
         CTEST_ERR("%s:%d  expected %" PRIdMAX "-%" PRIdMAX ", got %" PRIdMAX, caller, line, exp1, exp2, real);
+#else
+        CTEST_ERR("%s:%d  expected %d-%d, got %d", caller, line, exp1, exp2, real);
+#endif
     }
 }
 

